### PR TITLE
feat(Select): Extra render prop in Option

### PR DESCRIPTION
**Pull request by y-lohse**
I needed to add some buttons *inside* an `Option`. I doubt anyone will need the same buttons, so I added an option that allows us to render extra elements inside the Option. Please [see the new documentation for more details](https://yannick-lohse.fr/cozy-ui/react/#selectbox) ("Further control over options" section).

### CODE REVIEW

**Comment by y-lohse** [comment]
[(3) Groupes - supprimer](https://trello.com/c/AUW3POQY/1404-3-groupes-supprimer)

**Comment by Crash--** [comment]
I agree with you that we will not have the same buttons in our apps. But having a common wrapper for theses extra (with the right margin-right, flex alignment...) can be a good idea?

**Comment by y-lohse** [comment]
I'm going to add a pre-made wrapper around `renderExtra`, specifically to render a bunch of button-icon with proper styling. Hang tight.

**Comment by y-lohse** [comment]
Ok, I rewrote this PR. The original `Option` component is unchanged, and instead we have a new `ActionsComponent` that can be used to inject… well, actions into an option. Documentation and example is still here : https://yannick-lohse.fr/cozy-ui/react/#selectbox

File: react/SelectBox/Readme.md [hunk 1]
```diff
@@ -66,6 +66,28 @@ const options = [
...
+const ActionsOption = require('../SelectBox').ActionsOption;
+const options = [
+  { value: 'chocolate', label: 'Chocolate' },
+  { value: 'strawberry', label: 'Strawberry', isDisabled: true },
+  { value: 'vanilla', label: 'Vanilla' }
+];
+
+const CustomOption = (props) => (<ActionsOption {...props} actions={{
+  delete: () => alert('deleting'),
+  rename: ({ data }) => alert(data.value)
```
[the remark this answers is not in the document]
Comment (y-lohse) [reply react/SelectBox/Readme.md]: Yeah i wondered about that. The `SelectionBar` component also has silent icons so I wanted to keep it consistent, but I think i also like the more verbose option better.

File: react/SelectBox/SelectBox.jsx [hunk 2]
```diff
@@ -109,6 +113,34 @@ const CheckboxOption = ({ ...props }) => <Option {...props} withCheckbox />
...
+            action.handler(props)
+          }}
+        />
+      ))}
+    </span>
+  </Option>
+)
+
+ActionsOption.propTypes = {
+  actions: PropTypes.objectOf(PropTypes.func)
```
Comment (ptbrowne) [inline react/SelectBox/SelectBox.jsx]: The propTypes is no longer OK.

PropTypes.shape({
  icon: PropTypes.string,
  handler: PropTypes.func
})

File: react/SelectBox/SelectBox.jsx [hunk 3]
```diff
@@ -109,6 +113,34 @@ const CheckboxOption = ({ ...props }) => <Option {...props} withCheckbox />
...
+}
+
+ActionsOption.defaultProps = {
+  actions: {}
```
Comment (ptbrowne) [inline react/SelectBox/SelectBox.jsx]:
```suggestion
  actions: []
```

**Comment by ptbrowne** [comment]
LGTM after the small remarks. Thanks @y-lohse !

**Comment by ptbrowne** [comment]
One thing that bothers me a bit is the style. When nothing is selected, the action buttons float in the middle of nowhere.

image

I have trouble also understanding the specific usecase here. If I've understood well, these would be used in the context of a SelectBox to add/remove groups to a contact. In this case, why are there action buttons on the group ? As a user, I don't think my primary intention when opening this select box is to rename/delete a group. It clutter the UI a bit. @Claiw @joel-costa ?

File: react/SelectBox/Readme.md [hunk 4]
```diff
@@ -66,6 +66,28 @@ const options = [
...
+```
+const ActionsOption = require('../SelectBox').ActionsOption;
+const options = [
+  { value: 'chocolate', label: 'Chocolate' },
+  { value: 'strawberry', label: 'Strawberry', isDisabled: true },
+  { value: 'vanilla', label: 'Vanilla' }
+];
+
+const CustomOption = (props) => (<ActionsOption {...props} actions={[
+    { icon: 'delete', handler: () => alert('deleting') },
```
Comment (Crash--) [inline react/SelectBox/Readme.md]: `handler`? Since this is designed to be executed during the `onClick`, why not naming it `onClick` ?

Comment (y-lohse) [reply react/SelectBox/Readme.md]: That would work too — but it's also the handler for the action, or the event. I don't know, I don't think one is much better than the other ?

File: react/SelectBox/SelectBox.jsx [hunk 5]
```diff
@@ -109,6 +113,39 @@ const CheckboxOption = ({ ...props }) => <Option {...props} withCheckbox />
...
+      icon: PropTypes.string,
+      handler: PropTypes.func
+    })
+  )
+}
+
+ActionsOption.defaultProps = {
+  actions: []
+}
+
```
Comment (Crash--) [inline react/SelectBox/SelectBox.jsx]: New file? :-°

**Comment by Claiw** [comment]
hi, the screenshot I see doesn't fit with the specifications.
Here is the expected behaviour you can find in our zeplin spec artboard :
screen shot 2018-11-23 at 18 12 03
on the hover of an item the actions appears, there are not supposed to be displayed by default.

**Comment by ptbrowne** [comment]
I think the actions are displayed by default for the menu to work on mobile without any extra work. I think they could be showed by default on mobile and hidden on desktop, while we wait to display a proper action menu on mobile. @y-lohse ?

**Comment by y-lohse** [comment]
Yep, could work. Lemme try that.

**Comment by Claiw** [comment]
💯

**Comment by ptbrowne** [comment]
I created an issue [here](https://github.com/cozy/cozy-ui/issues/719) since this PR is closed :)

**Pull request merged**